omnigibson/prims/rigid_prim.py

```diff
@@ -605,6 +605,142 @@ def contact_reporting_enabled(self):
+        except scipy.spatial.qhull.QhullError:
+            # Handle the case where a convex hull cannot be formed (e.g., collinear points)
+            return None
+        
+    @cached_property
+    def visual_boundary_points(self):
+        """
+        Returns:
+            np.ndarray: points on the convex hull of all points from child geom prims
+        """
+        return self._compute_points_on_convex_hull(visual=True)
+    
+    @cached_property
+    def collision_boundary_points(self):
+        """
+        Returns:
+            np.ndarray: points on the convex hull of all points from child geom prims
+        """
+        return self._compute_points_on_convex_hull(visual=False)
+        
+    @property
+    def aabb(self):
+        position, orientation = self.get_position_orientation()
+        hull_points = self.collision_boundary_points
+        if hull_points is None:
+            # TODO: Decide if this is the right thing to do
+            return position, position
+        
+        scale = self.scale
+        points_scaled = hull_points * scale
+        points_rotated = np.dot(T.quat2mat(orientation), points_scaled.T).T
+        points_transformed = points_rotated + position
+
+        aabb_lo = np.min(points_transformed, axis=0)
+        aabb_hi = np.max(points_transformed, axis=0)
+        return aabb_lo, aabb_hi
+
+    @property
+    def aabb_extent(self):
+        """
+        Get this xform's actual bounding box extent
+
+        Returns:
+            3-array: (x,y,z) bounding box
+        """
+        min_corner, max_corner = self.aabb
+        return max_corner - min_corner
+
+    @property
+    def aabb_center(self):
+        """
+        Get this xform's actual bounding box center
+
+        Returns:
+            3-array: (x,y,z) bounding box center
+        """
+        min_corner, max_corner = self.aabb
+        return (max_corner + min_corner) / 2.0
+    
+    @property
+    def visual_aabb(self):
+        position, orientation = self.get_position_orientation()
+        scale = self.scale
+        points_to_transform = []
+
+        if self.visual_boundary_points is None:
+            # TODO: Decide if this is the right thing to do
+            # No predefined hull points, gather corners from mesh extents
+            for mesh in self._visual_meshes.values():
+                extent = mesh.prim.GetAttribute("extent").Get()
+                lo, hi = np.array(extent[0]), np.array(extent[1])
+                # Append corners of the bounding box
+                points_to_transform.extend([
+                    [lo[0], lo[1], lo[2]], [hi[0], lo[1], lo[2]],
+                    [lo[0], hi[1], lo[2]], [hi[0], hi[1], lo[2]],
+                    [lo[0], lo[1], hi[2]], [hi[0], lo[1], hi[2]],
+                    [lo[0], hi[1], hi[2]], [hi[0], hi[1], hi[2]]
+                ])
+        else:
+            # Use predefined hull points directly
+            points_to_transform.extend(self.visual_boundary_points)
```

Comment:
like mentioned before, we should just use all the points. By the way, the logic between visual_aabb and collision_aabb can be shared.

Reply:
will be fixed in https://github.com/StanfordVL/OmniGibson/pull/582